compiler/qsc_passes/src/baseprofck.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_hir::{
+    hir::{
+        BinOp, CallableDecl, CallableKind, Expr, ExprKind, Item, ItemKind, Lit, Package, SpecBody,
+        SpecGen, Stmt,
+    },
+    ty::{Prim, Ty},
+    visit::{walk_expr, walk_item, Visitor},
+};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
```

Comment:
There's a lot more changes to internals than I was expecting. Or is this just work from Stefan's  branch that isn't merged into main in those PRs yet?

Reply:
Ah, I see you mention 494 in the description. Never mind...